Mlem/Models/Content/Post/PostModel+MenuFunctions.swift

```diff
@@ -136,40 +157,107 @@ extension PostModel {
-                postTracker: postTracker
+        #if DEBUG
+            functions.append(
+                buildDeveloperMenu(
+                    editorTracker: editorTracker,
+                    postTracker: postTracker
+                )
             )
-        )
-#endif
+        #endif
         
         return functions
     }
+
     // swiftlint:enable function_body_length
     
-#if DEBUG
-    private func buildDeveloperMenu(
-        editorTracker: EditorTracker,
+    // swiftlint:disable:next function_body_length
+    private func modMenuFunctions(
+        community: CommunityModel,
+        modToolTracker: ModToolTracker,
         postTracker: StandardPostTracker?
-    ) -> MenuFunction {
-        MenuFunction.childMenu(
-            titleKey: "Developer Menu",
-            children: [
-                .standardMenuFunction(
-                    text: "Toggle Read State",
-                    imageName: "book.and.wrench",
-                    enabled: true,
-                    callback: {
-                        Task {
-                            let newState = !self.read
-                            await self.markRead(newState)
-                        }
-                    }
+    ) -> [MenuFunction] {
+        var functions: [MenuFunction] = .init()
+        
+        functions.append(MenuFunction.toggleableMenuFunction(
+            toggle: post.featuredCommunity,
+            trueText: "Unpin",
+            trueImageName: Icons.unpin,
+            falseText: "Pin",
+            falseImageName: Icons.pin
+        ) {
+            Task {
+                await self.toggleFeatured(featureType: .community)
+                await self.notifier.add(.success("\(self.post.featuredCommunity ? "P" : "Unp")inned post"))
+            }
+        })
+        
+        functions.append(MenuFunction.toggleableMenuFunction(
+            toggle: post.locked,
+            trueText: "Unlock",
+            trueImageName: Icons.unlock,
+            falseText: "Lock",
+            falseImageName: Icons.lock
+        ) {
+            Task {
+                await self.toggleLocked()
+                await self.notifier.add(.success("\(self.post.locked ? "L" : "Unl")ocked post"))
+            }
+        })
+        
+        functions.append(MenuFunction.toggleableMenuFunction(
+            toggle: post.removed,
+            trueText: "Restore",
+            trueImageName: Icons.restore,
+            falseText: "Remove",
+            falseImageName: Icons.remove,
+            falseRole: .destructive(prompt: nil)
+        ) {
+            modToolTracker.removePost(self, shouldRemove: !self.post.removed)
+        })
+
+        if creator.userId != siteInformation.userId {
+            functions.append(MenuFunction.toggleableMenuFunction(
+                toggle: creatorBannedFromCommunity,
+                trueText: "Unban author",
+                trueImageName: Icons.communityUnban,
+                falseText: "Ban author",
```

Comment:
We should use Title Case here. Also, we should standardise our wording of "User" vs "Author" - here, we use "Ban Author" but we use "Block User" in other places.

```suggestion
                trueText: "Unban Author",
                trueImageName: Icons.communityUnban,
                falseText: "Ban Author",
```

Reply:
Changed to "User"--I think "Author" is marginally more precise, but that's OOS for this PR